tests/integration/data_sources_and_expectations/expectations/test_expect_query_results_to_match_source.py

```diff
@@ -0,0 +1,326 @@
+import pandas as pd
+import pytest
+
+import great_expectations.expectations as gxe
+from great_expectations.expectations.metrics.util import MAX_RESULT_RECORDS
+from tests.integration.conftest import (
+    MultiSourceBatch,
+    MultiSourceTestConfig,
+    multi_source_batch_setup,
+)
+from tests.integration.data_sources_and_expectations.data_sources.test_source_to_target import (
+    ALL_SOURCE_TO_TARGET_SOURCES,
+)
+from tests.integration.test_utils.data_source_config import SqliteDatasourceTestConfig
+
+SQLITE_ONLY = [
+    MultiSourceTestConfig(
+        source=SqliteDatasourceTestConfig(),
+        target=SqliteDatasourceTestConfig(),
+    )
+]
+
+SOURCE_DATA = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+
+TARGET_DATA = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6], "c": [4, 5, 6]})
+
+SOURCE_DATA_WITH_DUPS = pd.DataFrame({"a": [1, 1, 2, 3], "b": [4, 4, 5, 6]})
+
+TARGET_DATA_WITH_DUPS = pd.DataFrame({"a": [1, 1, 2, 3], "b": [4, 4, 5, 6]})
+
+MAX_LENGTH_TARGET_DATA = pd.DataFrame(
+    {"a": [idx for idx in range(300)][100:], "b": [idx for idx in range(400)][200:]}
```

Comment:
Non-blocking, but the range + splicing took me an extra second to parse. I'd consider `[idx for idx in range(300)][100:]` -> `list(range(100, 300))`, etc.